source/agora/node/LederOutputStream.d

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+
+    Function definition and helper related to serialization
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.LedgerOutputStream;
+
+import agora.common.Transaction;
+
+/// Ditto
+public class LedgerOutputStream
```

Comment:
It seems that this whole class is re-inventing a GC-backed dynamic array.
The `write` methods use a pattern of `reserve` (which re-allocates the data), then write to a slice.
Instead, one can just use the appending operator (`~=`) to obtain the same effects. I commented on one of the methods with an example.

Additionally, we should try to use `struct` over `class` when there is no need for OOP.

Reply:
For example:

```D
auto bytes = cast(ubyte[long.sizeof]*)&value;
this.data ~= (*bytes)[];
```

Reply:
`s/auto/const/`

Reply:
What do you mean?

Reply:
`s/word1/word` is a common way of expressing substitution. So, just replace `auto bytes` with `const bytes` here.

Personally I don't think it matters when there's only two lines of code.

Reply:
Well actually it could matter if the backing store's element type is const I suppose. Anyway yeah, just maeke it const.